libgnucash/backend/dbi/gnc-backend-dbi.cpp

```diff
@@ -599,28 +616,29 @@ adjust_sql_options (dbi_conn connection)
 template <DbType Type> bool
 drop_database(dbi_conn conn, const UriStrings& uri)
 {
+    // The user may have CREATE privileges even if it does not have access to the "root_db"
     const char *root_db;
     if (Type == DbType::DBI_PGSQL)
     {
         root_db = "template1";
+	if (dbi_conn_select_db (conn, root_db) == -1)
+        {
+	    PERR ("Failed to switch out of %s, drop will fail.",
+		  uri.quote_dbname(Type).c_str());
+	    LEAVE ("Error");
+	    return false;
+	}
     }
     else if (Type == DbType::DBI_MYSQL)
     {
-        root_db = "mysql";
+        // There is no need to first switch to another DB
     }
     else
```

Comment:
You might as well do
```
else if (Type != DbType::DBI_MYSQL)
{
    PERR...
```
and change the comment to "Postgres can't drop a database it's connected to".